ui/src/app/core/settings.service.ts

```diff
@@ -0,0 +1,83 @@
+import {Injectable} from '@angular/core';
+import {AuthService} from '../core/auth.service';
+import {CapabilitiesService} from "./capabilities.service";
+
+type Settings = {
+  v1: {
+    // set to '' for basic auth users
+    userId: string
+    // set to '' for non-project-based job manager
+    projects: ProjectSettings[]
+  }
+};
+
+type ProjectSettings = {
+  projectId: string,
+  displayColumns: string[]
+  // ... other browser-based settings to be saved go here
+}
+
+@Injectable()
+export class SettingsService {
+  currentSettings: Settings;
+
+  /** If there are browser settings for the current user (or for '' if none) get them;
+   * otherwise, empty out the browser settings and set up a clean scaffolding */
+  constructor(
+    private readonly authService: AuthService,
+    private readonly capabilitiesService: CapabilitiesService,
+    private readonly localStorage: Storage
+  ) {
+    const savedSettings = JSON.parse(this.localStorage.getItem('settings'));
+    if ((authService.userId && savedSettings && (savedSettings.v1.userId == this.authService.userId))
+      || !authService.userId && savedSettings && savedSettings.v1.userId === '') {
+      this.currentSettings = savedSettings;
+    } else {
+      let currentUser = '';
+      if (authService.userId) {
+        currentUser = this.authService.userId;
+      }
+      this.currentSettings = {
+        'v1': {
+          'userId' : currentUser,
+          'projects': [{
+            'projectId': '',
+            'displayColumns': []
+          }]} };
+      this.updateLocalStorage();
+    }
+  }
+
+  /** return the display columns settings for a project, or if they don't exist, set all
+   * possible columns from capabilities service to be displayed and return that */
```

Comment:
this comment seems out of date, if the settings don't exist you're just returning `undefined` here (which the caller interprets as display all the columns with `primary = true`). Also, tiny nit, but prefer to start comments with a capital letter.

Reply:
fixed